apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,237 @@
+   * This will throw an axios error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param ids Array of hex-encoded price feed IDs for which updates are requested.
+   * @param encoding Optional encoding type. If true, return the price update in the encoding specified by the encoding parameter. Default is hex.
+   * @param parsed Optional boolean to specify if the parsed price update should be included in the response. Default is false.
+   * @returns Array of PriceFeed objects containing the latest updates.
+   */
+  async getLatestPriceUpdates(
+    ids: HexString[],
+    encoding?: EncodingType,
+    parsed?: boolean
+  ): Promise<PriceUpdate[]> {
+    const url = new URL(`${this.baseURL}/v2/updates/price/latest`);
+    // Append parameters to the URL search parameters
+    ids.forEach((id) => url.searchParams.append("ids[]", id));
+    if (encoding) {
+      url.searchParams.append("encoding", encoding);
+    }
+    if (parsed !== undefined) {
+      url.searchParams.append("parsed", String(parsed));
+    }
+    return await this.httpRequest(url.toString());
+  }
+
+  /**
+   * Fetch the price updates for a set of price feed IDs at a given timestamp.
+   * This endpoint can be customized by specifying the encoding type and whether the results should also return the parsed price update.
+   * This will throw an axios error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param publishTime Unix timestamp in seconds.
+   * @param ids Array of hex-encoded price feed IDs for which updates are requested.
+   * @param encoding Optional encoding type. If true, return the price update in the encoding specified by the encoding parameter. Default is hex.
+   * @param parsed Optional boolean to specify if the parsed price update should be included in the response. Default is false.
+   * @returns Array of PriceFeed objects containing the latest updates.
+   */
+  async getTimestampPriceUpdates(
+    publishTime: UnixTimestamp,
+    ids: HexString[],
+    encoding?: EncodingType,
+    parsed?: boolean
+  ): Promise<PriceUpdate[]> {
+    const url = new URL(`${this.baseURL}/v2/updates/price/${publishTime}`);
+    ids.forEach((id) => url.searchParams.append("ids[]", id));
+    if (encoding) {
+      url.searchParams.append("encoding", encoding);
+    }
+    if (parsed !== undefined) {
+      url.searchParams.append("parsed", String(parsed));
+    }
+    return await this.httpRequest(url.toString());
+  }
+
+  /**
+   * Fetch streaming price updates for a set of price feed IDs.
+   * This endpoint can be customized by specifying the encoding type, whether the results should include parsed updates,
+   * and if unordered updates or only benchmark updates are allowed.
+   * This will return an EventSource that can be used to listen to streaming updates.
```

Comment:
What is the behaviour if invalid ID is passed? Will it throw an error or?

Reply:
it will return error 404 `Price ids not found: xxxx` as per how it is if you directly access the hermes endpoint